docs/learn/advanced/security.md

```diff
@@ -0,0 +1,37 @@
+
+The *earliest anchor rule* solves any conflict in the history of a stream by picking the stream that was anchored at the earliest point in time. This enables keys to be securely revoked since someone that gains possession of an old key after it was revoked will be unable to produce a *proof-of-publication* that is earlier than the first anchor. A drawback with this strategy is that the controller of the stream can create anchored commits in secret and reveal them at a later point in time to change the history of the stream. However, this can be mitigated by allowing the log to have multiple branches of history and merge them using a CRDT. This strategy is used by the [TileDocument](../../streamtypes/tile-document/overview.md).
+
+The *latest nonce rule* solves any conflict by simply picking the signed commit which includes the largest nonce. This strategy doesn't support key revocation, so the controller of a stream with this strategy can't be changed. Ceramic doesn't support this strategy yet, but it is required for StreamTypes like [DIDPublish](https://github.com/ceramicnetwork/CIP/issues/105). In the future The Caip10Link StreamType may also be updated to use this strategy for efficiency reasons.
+
+### Network gossip
+Ceramic nodes use libp2p pubsub to gossip about stream tips. The two operations that happen in the gossip is publication of stream updates and queries for tips by nodes that load a specific stream. The main security consideration for a Ceramic node is that any new tip that comes in for a stream which it cares about could be a fake or invalid tip for that stream and thus needs to be verified.
+
+#### DoS attack
+A malicious node can spam the pubsub topic by sending a lot of messages. This may be randomly generated messages that are not even for valid streams, or it could be invalid tips for valid streams that get rejected by the stream update rules (the "false log attack" explained below). The main counter attack for this is to limit the amount of messages a single node can send by having an automated reputation system that disconnects from nodes which are spammy. No such reputation systems exists in ceramic today, though it is planned as a future improvement.
+
+#### False log attack
+A malicious node can spam nodes that pin specific streams by sending faulty commit logs. The unsuspecting node would sync the log of the stream and when it validates it the stream would get invalidated and removed from the local node. However, if the false log is long and the malicious node send multiple of these it may cause a significant amount of overhead. The most simple countermeasure to this is to simply stop accepting tips from nodes which have proven to not be reliable. A more significant approach would be to build a StreamType that includes a recursive zero-knowledge proof that the log is indeed correctly associated with the given streamid.
```

Comment:
```suggestion
A malicious node can spam nodes that pin specific streams by sending faulty commit logs. The unsuspecting node would sync the log of the stream and find it invalid, at which point the node would throw out the faulty log without applying it to its local copy of the stream's state. However, if the false log is long and the malicious node sends multiple of these it may cause a significant amount of overhead. The most simple countermeasure to this is to simply stop accepting tips from nodes which have proven to not be reliable. A more significant approach would be to build a StreamType that includes a recursive zero-knowledge proof that the log is indeed correctly associated with the given streamid.
```

Reply:
Reworded a bit, previous wording made it sound like a faulty log could cause a node to throw out all state for a stream, even the valid state it already had